client/verta/verta/_internal_utils/model_validator.py

```diff
@@ -95,4 +99,16 @@ def must_verta(model):
             " argument and return types may change unintuitively when deployed"
         )
 
+    expected_test_params = list(
+        inspect.signature(VertaModelBase.model_test).parameters.keys()
```

Comment:
This would always return `['self']` right?  Is the idea here to ensure that no arguments are passed to `model_test()`?

Reply:
This does return `['self']`, yes! And the implied idea _is_ to ensure no arguments are passed to `model_test()`.

But technically the core idea is to ensure that the user's implementation of `model_test()` has the same signature as the interface `VertaModelBase.model_test()` since Python itself doesn't have a way to enforce that, I don't think

Reply:
This does return `['self']`, yes! And the implied idea _is_ to ensure no arguments are passed to `model_test()`.

But technically the core idea is to ensure that the user's implementation of `model_test()` has the same signature as the interface `VertaModelBase.model_test()`.

Reply:
Hmm I see your point, but for me the more important thing is "implementation-wise, the function signature's parameters match" rather than "conceptually, `model_test()` doesn't accept arguments". And even that wording is confusing to me, since `self` is an argument.

To me, this error message is actually already as explicit as possible, in that it says exactly what parameters the method signature should have. 🤔

Reply:
Granted, the user likely arrives at the same conclusion whether it is via `['self'] != ['self', 'thing_you_added']` or `don't use args with this function`.  It's the precise language of "parameters must be" within a message intended to convey "don't use parameters here" that arguably could lead to confusion.  Point taken about `self` _technically_ being a parameter.  Honestly, I don't feel strongly enough to fight for a change here.  I just wanted to point out how another reasonable person might interpret it.

Reply:
haha this was helpful to point out—I wouldn't have considered it myself. If users start bringing it up, I'll make the change.